src/printer.js

```diff
@@ -562,7 +562,20 @@ function genericPrintNoParens(path, options, print) {
       parts.push("return");
 
       if (n.argument) {
-        parts.push(" ", path.call(print, "argument"));
+        if (n.argument.comments) {
+          parts.push(
+            n.loc.start.line === n.loc.end.line
+              ? concat([" ", path.call(print, "argument")])
```

Comment:
looks like this is the same as the else branch. Could you only have two branches instead of three?